### ISSUE TITLE
Support AV1

### ISSUE THREAD

**Issue by cromefire** [jellyfin/jellyfin-web#1241]
Support detection of av1 in firefox and chrome:

To probe support use:
```javascript
video.canPlayType("video/webm; codecs=av1");
```

**Comment by cromefire** [comment]
To try av1 in firefox (desktop), turn on "always prefer" [here](https://www.youtube.com/account_playback) and then try [this](https://www.youtube.com/watch?v=oAbB4dQOz4I&list=PLyqf6gJt7KuHBmeVzZteZUlNUQAVLwrZS) (ensure av1 is used in the stats for nerds)

**Comment by MrTimscampi** [comment]
We already do: #1167

**Comment by cromefire** [comment]
Okay but av1 in mp4 support is still missing. This would be `video/mp4;Codecs=av1`

**Comment by cromefire** [comment]
This it what YouTube for example does

**Comment by Artiume** [comment]
So av1 files can't be streamed in .ts containers. we have to use mp4 containers. I tried disabling video transcoding to verify it would work in the browser and it kept failing. Turns out it was trying to Direct Stream and ffmpeg ate itself. 

https://github.com/SPBTV/video_av1_samples

**Comment by cromefire** [comment]
Yeah webm and mp4 only and some weird format (ivf or so?) for now

**Comment by cromefire** [comment]
Not stale

**Comment by cromefire** [comment]
Shush bot

**Comment by Extarys** [comment]
Can we also add mkv containers? I know how to work with mkv and it's tools,etc. I must not be alone using mkv containers with av1? I tried streaming a mkv av1 50fps video in jellyfin and got a not supported error, quite sad.

It would feel weird having a bluray collection locally with a "webm" file format, or is it?
I used mp4 "back in the days" when the video was x264 encoded but that is pretty much it. all my videos are mkv now.

**Comment by MrTimscampi** [comment]
MKV is unfortunately not supported by browsers.

Historically, MKV was treated as WebM (By Emby, then that behavior was inherited by Jellyfin). However, we removed that behavior because, while WebM is essentially a small subset of MKV, it's not fully compatible with MKV, and caused some errors when files weren't compatible.

**Comment by Extarys** [comment]
@MrTimscampi Oh I see, thank you for the explanation. :heart: I guess it's easier in the backend to support webm only than trying to support both :disappointed: 

**It is a little off topic** so feel free to redirect me if needed, but I'll get faster answer and probably a better one too asking this here and now: If I want my library to be as jellyfin-friendly possible while having the flexibility and audio quality I currently have, should I start using webm for my new encodes with opus/av1 instead DTS and stuff? I mean, opus rocks. :v: 

x264 videos inside mkv should be pretty easy to "port", just need to transcode the audio track. Main issue is vodsub but I know there is plan to extract the images on the server and support them in browser or something so no biggy on that part. The sad thing is aas subtitles won't have the custom fonts because webm doesn't support attachments. Kodi and the mpv-shim can use those tho, but I understand the browser's limitation too.

**Comment by Nothing4You** [comment]
bad bot

**Comment by Extarys** [comment]
No, bot!

**Comment by Nothing4You** [comment]
*insert useless comment here that generates activity for the bot to see*

**Comment by LinAGKar** [comment]
>So av1 files can't be streamed in .ts containers. we have to use mp4 containers

Same goes for FLAC. I can directstream AV1+FLAC to firefox just fine if it's in MP4, but if it's in MKV it will be transcoded into H.264+AAC to be put into a TS container.

**Comment by LinAGKar** [comment]
> hi everyone ! did you notice Firefox support AV1 codec since rev 100 ? :)

It's supported it for a lot longer than that

**Comment by LinAGKar** [comment]
@LinAGKar Not sure what you're trying to say, but Firefox has supported AV1 since version 67 (and since 65 on some platforms). Though Firefox 100 added support for AV1 hardware decoding on Windows.

**Comment by gaalos** [comment]
@LinAGKar thx for your reply.
i convert Movie to AV1 but when i read it on firefox, jellyfin convert it to 264

**Comment by Extarys** [comment]
@gaalos does it has DTS audio track of VobSub subtitles? If so, it'll need transcoding I believe.

**Comment by gaalos** [comment]
@Extarys 
![image](https://user-images.githubusercontent.com/32361668/172920382-5defae83-078e-4555-a38a-bb5b37296413.png)

**Comment by nyanmisaka** [comment]
Firefox doesn’t support EAC3 audio.

**Comment by gaalos** [comment]
@nyanmisaka thx for your reply. 
Apprently OPUS is better than AAC as same biterate. 
AV1 is made for low bitrate with good quality.
Make sens to use OPUS with.

**Comment by Extarys** [comment]
I tend to convert DTS and EAC3 to Opus. Usually 128kb per 2 channels, a little more if it's DTS and I really want to preserve more quality. I know about generation loss and all, but I prefer to direct stream instead of transcoding on the fly the whole video every time.

**Comment by gaalos** [comment]
@Extarys did you use AV1 with your Opus format ?

**Comment by Extarys** [comment]
@gaalos Yes, and when I have VobSub as subtitles I disable the subtitles before playing the episode.

The Jellyfin desktop client would work fine with EAC3 audio, but doesn't support AV1 lol

**Comment by gaalos** [comment]
@Extarys, i don't have issues with desktop client But i have with Android TV Client 

[Issues link](https://github.com/jellyfin/jellyfin-androidtv/issues/1768)

I upload some videos test if you want test it @Extarys  [Here](https://wetransfer.com/downloads/d3e755eb89fbc3b624e22a3850722ab720220609225107/03ff4e/grid)

You can found AV1 with OPUS and AV1 with EAC3, i don't have issues except on android TV

**Comment by Extarys** [comment]
I do not own any Android TV client to try and replicate your current issue. Perhaps another member of the community will be able to test and help provide logs.

**Comment by DesertCookie** [comment]
> I upload some videos test if you want test it @Extarys [Here](https://wetransfer.com/downloads/d3e755eb89fbc3b624e22a3850722ab720220609225107/03ff4e/grid)

I'd be happy to test on my Shield TV if you were to reupload the files.

**Comment by D33M0N** [comment]
10.10.7 ... playing AV1 in mkv still fails in both Firefox and LG TV. And yet... TV does play _SOME_ AV1's in mkv just fine... it's like there are some different AV1 formats??

**Comment by thornbill** [comment]
AV1 in MP4 should work in compatible browsers now.
